web/client/src/components/form/TokenCredentials.jsx

```diff
@@ -67,11 +76,33 @@ export default function TokenCredentials(props) {
     setFieldValue("tokenCredentials", str);
   };
 
+  const toggleSave = () => {
+    setFieldValue("saveTokenLocally", !saveCredential);
+    setSaveCredential((prev) => !prev);
+  };
+
+  const saveTokenToggle = () => {
+    return (
+      <FormControlLabel
+        className="form-save-cred-toggle"
+        control={
+          <Switch
+            checked={saveCredential}
+            onChange={toggleSave}
+            color="primary"
+          />
+        }
+        label="Save credential for future use"
```

Comment:
credentials (for all user facing usage of this word, use the plural form to be consistent)